### ISSUE TITLE
Trailing slashes added from registered routes 

### ISSUE THREAD

**Issue by mhagnumdw** [pippo-java/pippo#525]
Hi!

```java
@Path("/webapp/api/p/releasenotes")
public class ReleaseNotesNgController {
    
    @PUT({ "", "/{id: \\d+}" })
    public void save(@Body ReleaseNotesFormDTO dto) {
        //....
    }

}
```

The routes registered for the above example are:
- Add route for PUT '/webapp/api/p/releasenotes/' **(with trailing slash)**
- Add route for PUT '/webapp/api/p/releasenotes/{id: \d+}'

But I believe they should be:
- Add route for PUT '/webapp/api/p/releasenotes' **(no trailing slash)**
- Add route for PUT '/webapp/api/p/releasenotes/{id: \d+}'

What do you think?

**Comment by munendrasn** [comment]
@mhagnumdw are you by any chance using `TrailingSlashHandler`?

**Comment by mhagnumdw** [comment]
> @mhagnumdw are you by any chance using `TrailingSlashHandler`?

I'm not using this feature.

**Comment by mhagnumdw** [comment]
Below the Pippo logs

```log4j
2019-10-13 10:39:05.947 DEBUG ro.pippo.core.route.DefaultRouter:230 (main) - Add route for PUT '/webapp/api/p/releasenotes/'
2019-10-13 10:39:05.947 DEBUG ro.pippo.core.route.DefaultRouter:230 (main) - Add route for PUT '/webapp/api/p/releasenotes/{id: \d+}'
```

**Comment by mhagnumdw** [comment]
The problem is exactly on this line [ControllerRegistry.createControllerRoutes](https://github.com/pippo-java/pippo/blob/master/pippo-controller-parent/pippo-controller/src/main/java/ro/pippo/controller/ControllerRegistry.java#L219)

When concatenating `"/webapp/api/p/releasenotes"` with `""` (empty) results in `/webapp/api/p/releasenotes/`

**Comment by decebals** [comment]
I think that I encountered something similar with `RouteGroup`. I will try to replicate the problem with `RouteGroupe` and I will come with details and a propose. I think that is the same problem.

**Comment by munendrasn** [comment]
@decebals @mhagnumdw I have added fix in the PR #544. It is a bit verbose but handles all cases. Let me know your thoughts on the fix
